src/dune_rules/lib.ml

```diff
@@ -1305,18 +1351,34 @@ end = struct
       in
       let runtime_deps =
         let* pps = pps in
-        Resolve.List.concat_map pps ~f:(fun pp ->
-            let open Resolve.O in
-            let* ppx_runtime_deps = pp.ppx_runtime_deps in
+        Resolve.Memo.List.concat_map pps ~f:(fun pp ->
+            let* ppx_runtime_deps =
+              let open Memo.O in
+              match db.host with
+              | None -> ppx_runtime_deps pp
+              | Some _host ->
+                (* (ppx_runtime_libraries ...) run in the target context, so
+                   these dependencies need to be resolved here rather than at
+                   instantiation of the ppx library (in the host context). *)
+                let+ resolved = resolve_ppx_runtime_deps db pp ~private_deps in
```

Comment:
I would expect the stack frame to be added before the call to `resolve_ppx_runtime_deps`